compiler/src/dotty/tools/dotc/transform/CheckUnused.scala

```diff
@@ -401,16 +410,27 @@ object CheckUnused:
      * The optional name will be used to target the right import
      * as the same element can be imported with different renaming
      */
-    def registerUsed(sym: Symbol, name: Option[Name], isDerived: Boolean = false)(using Context): Unit =
-      if !isConstructorOfSynth(sym) && !doNotRegister(sym) then
-        if sym.isConstructor && sym.exists then
-          registerUsed(sym.owner, None) // constructor are "implicitly" imported with the class
+    def registerUsed(sym: Symbol, name: Option[Name], notForImport: Boolean = false, isDerived: Boolean = false)(using Context): Unit =
```

Comment:
I find the logic harder to follow because of the `not` in `notForImport`, can we flip the value of this flag to make it mean "possibly form an import"?